zerver/lib/notifications.py

```diff
@@ -372,25 +377,12 @@ def handle_missedmessage_emails(user_profile_id, missed_email_events):
             msg_list.extend(get_context_for_message(msg))
 
     # Send an email per recipient subject pair
-    if user_profile.realm.domain == 'zulip.com':
-        for recipient_subject, msg_list in messages_by_recipient_subject.items():
-            unique_messages = {m.id: m for m in msg_list}
-            do_send_missedmessage_events_reply_in_zulip(
-                user_profile,
-                list(unique_messages.values()),
-                mesage_count_by_recipient_subject[recipient_subject],
-            )
-    else:
-        all_messages = [
-            msg_
-            for msg_list in messages_by_recipient_subject.values()
-            for msg_ in msg_list
-        ]
-        unique_messages = {m.id: m for m in all_messages}
-        do_send_missedmessage_events(
+    for recipient_subject, msg_list in messages_by_recipient_subject.items():
+        unique_messages = {m.id: m for m in msg_list}
+        do_send_missedmessage_events_reply_in_zulip(
             user_profile,
             list(unique_messages.values()),
-            len(messages),
+            mesage_count_by_recipient_subject[recipient_subject],
```

Comment:
(I guess that should probably be its own commit in this PR, since the type was present before)